zcash_client_backend/src/data_api/chain.rs

```diff
@@ -150,7 +258,8 @@ where
     })?;
 
     // Fetch the ExtendedFullViewingKeys we are tracking
-    let extfvks = data.get_extended_full_viewing_keys(params)?;
+    let extfvks = data.get_extended_full_viewing_keys()?;
```

Comment:
Ditto (what keeps `params` and `data` in-sync?)

Reply:
So, this is a wart that I'm not certain how to best address. The fact that the backend uses the parameters is not something that should be exposed in the type, because there can easily be backends that don't make use of them. But that leaves this hole where the two could get out of sync. I guess that one counterargument to the change I've made is that one might want a backend to be able to store data for multiple networks at once; in that case, though, every method would probably need to take the `params` argument, which is definitely unergonomic.

Reply:
The backend _must_ be network-aware, even if the type system doesn't reflect it (how else can keys and addresses be serialized?)

If we don't want the type system to reflect the network, then the alternative would be to expose the parameters _through_ the backend, by adding e.g. a `WalletRead::network` method.

Reply:
So, the in-memory backend is a good example - it doesn't do any serialization, so it doesn't need network information. Other backends, like the existing one, must be network-aware. I think that the current API reflects this separation properly.

Reply:
That's a really good point. Can we do that as a separate change subsequent to this PR? I think that will be fairly invasive, and I don't want to extend the scope any more here.

Reply:
Yes, this kind of refactor should be a separate PR.